pkg/virt-api/api.go

```diff
@@ -926,13 +930,15 @@ func (app *virtAPIApp) setupTLS(fs Filesystem) error {
 	app.clientCAFile = filepath.Join(app.certsDirectory, "/clientCA.crt")
 
 	// Write the certs to disk
-	err := fs.WriteFile(app.clientCAFile, app.clientCABytes, 0600)
-	if err != nil {
-		return err
+	if len(app.clientCABytes) != 0 {
+		err := fs.WriteFile(app.clientCAFile, app.clientCABytes, 0600)
+		if err != nil {
+			return err
+		}
 	}
 
 	if len(app.requestHeaderClientCABytes) != 0 {
-		f, err := fs.OpenFile(app.clientCAFile, os.O_APPEND|os.O_WRONLY, 0600)
+		f, err := fs.OpenFile(app.clientCAFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0600)
```

Comment:
It's unclear to me if what I'm about to point out matters much, but I'll say it anyways.  

previously the logic ensured we always clobber the clientCAFile and overwrite it.  Now that the client-ca is optional, it's possible we just append to the file if it doesn't already exist.  Maybe there's a restart/recovery scenario where the file already exists somehow before hand and we keep appending CAs to it (probably the same CA over and over)

It wouldn't hurt to reverse the ordering that the file is written to now and overwrite the file with the request-header-ca because it has to exist, then allow appending the client-ca.

Reply:
fixed by using requestheader CA only, which will always overwrite